### PR TITLE
Fix bug where mention balloon doesn't close after making a selection

### LINKED ISSUE #7573: Bug with @-autocomplete in EAF
After selecting the username with Enter or Tab, the selection doesn't go away. It doesn't replicate on LW

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/628521446211730/1205056749705176) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by oetherington**
We currently have a bug where the mentions balloon doesn't close after making a selection. This is replicatable on EAForum and LessWrong, with Algolia or Elasticsearch. I initially thought it was a regression from #7564 but reverting that made no difference.

Upon further inspection the bug appears to be a result of changes we made when vendoring the plugin, but that would suggest the bug existed for a long time. It's possible that the bug existed previously, but appeared to work because of some other subtle interaction that has changed recently.

The issue is that the code that checks if the cursor is placed after an already existing mention checks for a node with the "mention" attribute, but when we vendored the library we removed the "mention" attribute and instead choose to insert a normal link with a "linkHref" attribute, and the links we insert have a "?mention=" query parameter.

Fixes #7573

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205067076393365) by [Unito](https://www.unito.io)

### CODE REVIEW

File: public/lesswrong-editor/src/ckeditor5-mention/src/mentionui.js [hunk 1]
```diff
@@ -816,9 +850,9 @@ function isPositionInExistingMention( position ) {
   // See https://github.com/ckeditor/ckeditor5-engine/issues/1723.
   const hasMention = position.textNode && position.textNode.hasAttribute( 'mention' );
```
Comment (jpaddison3) [inline public/lesswrong-editor/src/ckeditor5-mention/src/mentionui.js:851]: Should we be changing this as well?

**Review by jpaddison3 — COMMENTED** [review]
Gonna wait on approve till I understand this better, have asked a question.

File: public/lesswrong-editor/src/ckeditor5-mention/src/mentionui.js (hunk 1 above)
Comment (oetherington) [reply public/lesswrong-editor/src/ckeditor5-mention/src/mentionui.js]: Oh yeah, good call

**Comment by oetherington** [comment]
That's the code path that displays the popup. That condition checks if the cursor is in/after an already existing mention and if it is then we early exit and don't display the popup (hence why that condition being broken leads to this bug).

**Comment by jpaddison3** [comment]
Yeah I got that. I guess the part that I need explained like I'm 5 is why there's a difference between focus and markerPosition and why the behavior needs to be different in each case.

**Comment by jpaddison3** [comment]
@oetherington

File: public/lesswrong-editor/src/ckeditor5-mention/src/mentionui.js [hunk 2]
```diff
@@ -807,6 +807,40 @@ function createFeedCallback( feedItems ) {
   };
 }
 
+function getNonWhitespaceNodeBefore(position) {
+  let nodeBefore = position.nodeBefore;
```
Comment (jpaddison3) [inline public/lesswrong-editor/src/ckeditor5-mention/src/mentionui.js:811]: This file has a terrible tabs and spaces problem

**Comment by jpaddison3** [comment]
Ok, right, you're not really changing any of the logic that deals with, "wtf even is this marker vs focus crap". You're just saying, there are some places where we were doing a faulty check for "is this node a mention", and correcting that code. Checks out.

**Review by jpaddison3 — APPROVED** [review]

**Comment by jpaddison3** [comment]
Nice work

**Pull request merged**